streams/src/main/java/org/apache/kafka/streams/kstream/internals/graph/TableProcessorNode.java

```diff
@@ -62,21 +59,17 @@ public String toString() {
     @SuppressWarnings("unchecked")
     @Override
     public void writeToTopology(final InternalTopologyBuilder topologyBuilder) {
+        processorParameters.addProcessorTo(topologyBuilder, parentNodeNames());
+
         final String processorName = processorParameters.processorName();
-        topologyBuilder.addProcessor(processorName, processorParameters.processorSupplier(), parentNodeNames());
 
         if (storeNames.length > 0) {
+            // todo(rodesai): remove me once all operators have been moved to ProcessorSupplier
             topologyBuilder.connectProcessorAndStateStores(processorName, storeNames);
         }
 
-        final KTableSource<K, V> tableSource =  processorParameters.processorSupplier() instanceof KTableSource ?
-                (KTableSource<K, V>) processorParameters.processorSupplier() : null;
-        if (tableSource != null) {
-            if (tableSource.materialized()) {
-                topologyBuilder.addStateStore(Objects.requireNonNull(storeFactory, "storeFactory was null"),
-                                              processorName);
-            }
-        } else if (storeFactory != null) {
+        if (storeFactory != null) {
+            // todo(rodesai) remove when KTableImpl#doFilter, KTableImpl#doTransformValues moved to ProcessorSupplier
```

Comment:
added these to the todo list in the jira but I'm assuming you're just going to knock these out after this PR?